benchmarks/benchmark_runner.hpp

```diff
@@ -258,24 +258,27 @@ struct BenchmarkRunnerGemm {
       block_C.emplace_back();
     }
 
-    for (int i=0; i < count; i++) {
-      block_A[i].reset(size_A);
-      block_B[i].reset(size_B);
-      block_C[i].reset(size_C);
-      initialize_block(block_A[i], seed + i);
-      initialize_block(block_B[i], seed + i);
-      initialize_block(block_C[i], seed + i);
+    try {
+      for (int i = 0; i < count; i++) {
+        block_A[i].reset(size_A);
+        block_B[i].reset(size_B);
+        block_C[i].reset(size_C);
+        initialize_block(block_A[i], seed + i);
+        initialize_block(block_B[i], seed + i);
+        initialize_block(block_C[i], seed + i);
+      }
+
+      block_D.reset(size_C);
+      block_ref_D.reset(size_C);
+    } catch (std::exception const &e) {
+      state.SkipWithError("Problem size too large for device.");
```

Comment:
```suggestion
      state.SkipWithError("Memory Allocation and initialization failed.");
```
You don't know that the size is the problem

Reply:
maybe worth adding the `e.what()` string to the error